### DISCUSSION TITLE
how use recompile button function as 'python' or 'bpy'

### DISCUSSION THREAD

**Discussion by warlockhb** [bnpr/Malt#472] [Q&A] [answered]
<img width="1910" alt="image" src="https://user-images.githubusercontent.com/71178397/212576107-f504afef-9859-4bdc-8a66-9f0a67cb3045.png">

<img width="542" alt="image" src="https://user-images.githubusercontent.com/71178397/212576132-16378d98-0dd8-425c-992a-1be38cc2f509.png">

I'm working on a panel with Python.
Whenever the panel value is updated, the parameter of the MALT shader is set with the panel value.
However, in this case, you need to run the recompile button.

For convenience and automation, is there a way to run this function from a script?

**Comment by pragma37** [comment, marked as answer]
Here's the code for the header UI:
https://github.com/bnpr/Malt/blob/Development/BlenderMalt/MaltNodes/MaltNodeTree.py#L623

So, `node_tree.update_ext(force_update=True)`.

**Reply by warlockhb** [reply]
node_tree = bpy.data.node_groups['Node Name']
node_tree.update_ext(force_update=True)

Thnaks for you! The biggest problem has been solved.😄
